src/common/transformations/src/transformations/common_optimizations/fuse_rotary_positional_embeddings.cpp

```diff
@@ -427,6 +428,79 @@ ov::pass::RoPEFusionPreprocess::RoPEFusionPreprocess() {
     this->register_matcher(m, callback);
 }
 
+ov::pass::RoPEFusionVIT3D::RoPEFusionVIT3D() {
```

Comment:
This pattern matching is almost the same with `RoPEFusionGPTNEOX`, the only difference is that the rank of input is changed from 4 to 3, do you think we could merge them together ? Just like `RoPEFusionQwen(int split_output_id)` we could change to `RoPEFusionGPTNEOX(int rank)`

Reply:
If not easy to do now, we can leave this as an improvement.

Reply:
@zhangYiIntel  Yes, the transformation code is copy from RoPEFusionGPTNEOX and just modify the rank from 4 to 3, but I don't know the background of RoPEFusionGPTNEOX, so I didn't merge them together, I think we need a review from transformation team.

Reply:
The `RoPEFusionGPTNEOX` is copied from CPU previous internal ROPE transformation, it has no special requirement but only describes the vanilla Rope with RotateHalf

Reply:
already merged them